core/commands/partials/dedicated_configurator.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require_relative '../../models/result'
+require_relative '../../services/machine_configurator'
+require_relative '../../models/network_settings'
+require_relative 'dedicated_configuration_generator'
+require_relative '../destroy_command'
+require_relative '../../services/network_checker'
+require_relative '../../services/product_attributes'
+require_relative '../../services/configuration_generator'
+require 'workers'
+
+# The configurator brings up the configuration for the dedicated machines
+class DedicatedConfigurator
+  def initialize(config, env, logger)
+    @config = config
+    @env = env
```

Comment:
Do not store reference to the `env`. Extract all needed services in constructor.